src/sensors/rev_encoder.hpp

```diff
@@ -1,16 +1,34 @@
+#include <cmath>
 #include <FreqMeasureMulti.h>
 
 #ifndef REV_ENCODER_H
 #define REV_ENCODER_H
 
-class RevEncoder {
+/// @brief the class for the Rev Through Bore Encoder(www.revrobotics.com/rev-11-1271/)
+class RevEncoder 
+{
 	private:
-		uint8_t pin;
+		/// @brief the pin number that the encoders signal pin is plugged into
+		uint8_t inPin;
+		/// @brief Used to read rise time of the encoder
 		FreqMeasureMulti freq;
+		//// @brief measure of current angle in ticks (1024 ticks = 1 rotation). When set to 1 individual readings are returned
+		float ticks;
+		/// @brief measure of current angle in radians 
+		float radians;
 	public:
-		RevEncoder(uint8_t pin);
-		int getRawOutput();
-		float getAngle();
+		/// @brief Construct a new rev_encoder object
+		/// @param encPin the pin number that the encoders signal pin is plugged into
+		/// @param baudrate the desired baudrate for the encoder
+		RevEncoder(uint8_t encPin, int baudrate);
+		/// @brief updates the current angle in ticks (1024 = 1 Rotation) and radians.
+		void read();
+		/// @brief get the last angle of the encoder ticks
+		/// @return the last angle of the encoder in ticks (1024 = 1 Rotation).
+		float getAngleTicks();
+		/// @brief get the last angle of the encoder in radians
+		/// @return the last angle of the encoder in radians
+		float getAngleRadians();
```

Comment:
What is the angle (tick or radian) measured from?

- What is 0 rotation?
- What value represents one full revolution?
- What happens after a full revolution? Does it keep counting up, jump back to 0, etc?
- Can there be negative numbers? Does positive values represent clockwise or counterclockwise rotation?

(if each of these questions can be answered in the documentation that'd make using your values much easier.

Reply:
After the discussion, you could answer these questions by describing something like:
on your get_ticks, returns in the range of [0,1023], where 1024 ticks = 2pi radians, (wrapped between revolutions)
get_radians returns in range [0,2pi] radians (wrapped between revolutions)